perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -708,3 +731,127 @@ def SetupProxy(self):
 
     if commands:
       self.RemoteCommand(";".join(commands))
+
+
+class ContainerizedDebianMixin(DebianMixin):
+  """Class representing a Virtual Machine that runs Remote
+  Commands within a Docker Container running Ubuntu"""
+
+  def PrepareVMEnvironment(self):
+    """Initializes docker before proceeding with preparation."""
+    self.RemoteHostCommand('mkdir -p %s' % vm_util.VM_TMP_DIR)
+    self.Install('docker')
+    self.InitDocker()
+    super(ContainerizedDebianMixin, self).PrepareVMEnvironment()
+
+  def InitDocker(self):
+    """Initializes the docker container daemon."""
+    init_docker_cmd = ('sudo docker run -d '
+                       '--net=host '
+                       '--workdir=%s '
+                       '-v %s:%s ' % (CONTAINER_WORK_DIR,
+                                      vm_util.VM_TMP_DIR,
+                                      CONTAINER_MOUNT_DIR))
+    for sd in self.scratch_disks:
+      init_docker_cmd += '-v %s:%s ' % (sd.mount_point, sd.mount_point)
+    init_docker_cmd += '%s sleep infinity ' % UBUNTU_DOCKER_IMAGE
+
+    resp, _ = self.RemoteHostCommand(init_docker_cmd)
+    self.docker_id = resp[:-1]
+    return self.docker_id
+
+  def RemoteCommand(self, command,
+                    should_log=False, retries=SSH_RETRIES,
+                    ignore_failure=False, login_shell=False,
+                    suppress_warning=False):
+    """Runs a command inside the container.
+
+    Args:
+      command: A valid bash command.
+      should_log: A boolean indicating whether the command result should be
+          logged at the info level. Even if it is false, the results will
+          still be logged at the debug level.
+      retries: The maximum number of times RemoteCommand should retry SSHing
+          when it receives a 255 return code.
+      ignore_failure: Ignore any failure if set to true.
+      login_shell: Run command in a login shell.
+      suppress_warning: Suppress the result logging from IssueCommand when the
+          return code is non-zero.
+
+    Returns:
+      A tuple of stdout and stderr from running the command.
+    """
+    # Escapes bash sequences
+    command = command.replace('\'', '\'\\\'\'')
```

Comment:
Using " for the outside quotes instead of ' would make this more readable - see https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Strings#Strings